#
# old_revision [f01a8a459beecb889ec21a8a33ebd0cd153d3e39]
#
# patch "libpurple/protocols/jabber/parser.c"
#  from [9ab56d28740cacf167e31733831b10a7e92a41c3]
#    to [ebf572c0b2cb2232f0749040cccdbb0f4eddd7dd]
#
============================================================
--- libpurple/protocols/jabber/parser.c	9ab56d28740cacf167e31733831b10a7e92a41c3
+++ libpurple/protocols/jabber/parser.c	ebf572c0b2cb2232f0749040cccdbb0f4eddd7dd
@@ -132,6 +132,18 @@ jabber_parser_element_text_libxml(void *
 	xmlnode_insert_data(js->current, (const char*) text, text_len);
 }
 
+static void
+jabber_parser_structured_error_handler(void *user_data, xmlErrorPtr error)
+{
+	JabberStream *js = user_data;
+
+	purple_debug_error("jabber", "XML parser error for JabberStream %p: "
+								 "Domain %i, code %i, level %i: %s\n",
+					   js,
+					   error->domain, error->code, error->level,
+					   (error->message ? error->message : "(null)"));
+}
+
 static xmlSAXHandler jabber_parser_libxml = {
 	NULL,									/*internalSubset*/
 	NULL,									/*isStandalone*/
@@ -164,7 +176,7 @@ static xmlSAXHandler jabber_parser_libxm
 	NULL,									/*_private*/
 	jabber_parser_element_start_libxml,		/*startElementNs*/
 	jabber_parser_element_end_libxml,		/*endElementNs*/
-	NULL									/*serror*/
+	jabber_parser_structured_error_handler	/*serror*/
 };
 
 void
@@ -179,6 +191,7 @@ void jabber_parser_free(JabberStream *js
 
 void jabber_parser_free(JabberStream *js) {
 	if (js->context) {
+		xmlSetStructuredErrorFunc(NULL, jabber_parser_structured_error_handler);
 		xmlParseChunk(js->context, NULL,0,1);
 		xmlFreeParserCtxt(js->context);
 		js->context = NULL;
@@ -187,6 +200,8 @@ void jabber_parser_process(JabberStream 
 
 void jabber_parser_process(JabberStream *js, const char *buf, int len)
 {
+	xmlSetStructuredErrorFunc(NULL, jabber_parser_structured_error_handler);
+
 	if (js->context ==  NULL) {
 		/* libxml inconsistently starts parsing on creating the
 		 * parser, so do a ParseChunk right afterwards to force it. */
